pytheus/mmap.py

```diff
@@ -0,0 +1,344 @@
+import abc
+import collections
+import mmap
+import os
+import struct
+import typing
+
+# See https://docs.python.org/3/library/struct.html#format-characters
+StructFormatString = typing.Literal["i", "d", "?", "s"]
+StructFormatType = typing.Union[int, float, bool, str]
+
+STRUCT_INT_FORMAT: StructFormatString = "i"
+STRUCT_FLOAT_FORMAT: StructFormatString = "d"
+STRUCT_BOOL_FORMAT: StructFormatString = "?"
+STRUCT_STR_FORMAT: StructFormatString = "s"
+
+BYTE_ALIGNMENT = 4
+
+DEFAULT_BACKEND = "DEFAULT_BACKEND"
+
+
+class MmapDictTypeBackend(abc.ABC):
```

Comment:
This class let's you read and write to an MMAP file, using `struct.pack` and `struct.unpack` to serialize/deserialize different binary types like integer, doubles and bools. For `str` we simply use `encode` and `decode`.